src/cascade/core/db.py

```diff
@@ -0,0 +1,55 @@
+from contextlib import contextmanager
+
+try:
```

Comment:
I don't think I have any try's in my imports, so this is making me wonder if/when I should.

Reply:
In general you shouldn't. I'm actually not 100% it's the best way of doing it here though this is a relatively common pattern. What I'm trying to do is make the dependency on ezfuncs optional. This allows test code or other systems that want to use this module to import it without actually having to install any of the central comp stuff. But if you then go and do something that actually does require ezfuncs you'll get an immediate error explaining that it isn't installed.

The other common way of doing this is to have the import inside the functions where it's actually used. But that often leads to having several copies of the same import scattered throughout the module, which is ugly and confusing.